Adding new hassio modulue

drwahl opened the pull request:
Initial commit for a new HomeAssistant.io module to report the state of
entities from hassio.

Signed-off-by: David Wahlstrom <david.wahlstrom@gmail.com>

drwahl committed:
Adding new hassio modulue

Initial commit for a new HomeAssistant.io module to report the state of
entities from hassio.

Signed-off-by: David Wahlstrom <david.wahlstrom@gmail.com>

enkore commented:
Looks good, but I think you need to add a dependency mock, see https://i3pystatus.readthedocs.io/en/latest/module.html#handling-dependencies

drwahl committed:
Fix some copy/pasta misses

Signed-off-by: David Wahlstrom <david.wahlstrom@gmail.com>

drwahl committed:
Add requests dep docstring note

Signed-off-by: David Wahlstrom <david.wahlstrom@gmail.com>

drwahl committed:
Add json to mock modules list

In order to retain sanity when dealing with json data in the hassio
test, the test uses the json library to manipulate a dict for the test.
As such, json needs to be include MOCK_MODULES.

Signed-off-by: David Wahlstrom <david.wahlstrom@gmail.com>

drwahl committed:
Revert "Add json to mock modules list"

Reverting this, as MOCK_MODULES is only non-std-lib modules, and json is
apart of the std lib.

This reverts commit b748b29e2b0bc322179915c3509abf1a76979699.

drwahl commented:
I'm having a tough time getting this CI job to pass. It's complaining about the "requests" module, but I added a bit to the docstring about needing it, and it's already included in MOCK_MODULES. What am I missing?

drwahl committed:
Remove unused MagicMock import

Signed-off-by: David Wahlstrom <david.wahlstrom@gmail.com>

drwahl committed:
add requests dep docstring to hassio test

Signed-off-by: David Wahlstrom <david.wahlstrom@gmail.com>
